pkg/config/crd/store.go

```diff
@@ -128,6 +135,7 @@ func (s *Store) Init(ctx context.Context, kinds []string) error {
 			glog.V(3).Infof("Failed to obtain resources for CRD: %v", err)
 			continue
 		}
+		s.cacheMutex.Lock()
```

Comment:
use defer. It protects from panics.

Reply:
I'm hesitant to use Lock here. This is in a loop, the use of defer here requires an extra func. Extra unnamed func just for defer doesn't look great to me.

Reply:
Do we need to re-check in a tight loop here? agreed that locking repeatedly in a tight loop is not very good.
Can we wait for 500ms or so before checking again ?

Reply:
added the delay.